packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -130,7 +130,7 @@ export abstract class AbstractPaymentService
   public abstract updatePayment(
     paymentSessionData: PaymentSessionData,
     context: Cart & PaymentContext
-  ): Promise<PaymentSessionResponse>
+  ): Promise<PaymentSessionResponse | PaymentSessionResponse["session_data"]>
```

Comment:
suggestion: Would be good to add a TODO or comment here to explain why this is happening. Without context, its confusing.